saleor/invoice/emails.py

```diff
@@ -1,14 +1,16 @@
 from templated_email import send_templated_mail
 
+from ..account.models import User
 from ..celeryconf import app
 from ..core.emails import get_email_context
+from ..invoice import events
+from ..plugins.manager import get_plugins_manager
 from .models import Invoice
 
 
-def collect_invoice_data_for_email(invoice_pk, template):
+def collect_invoice_data_for_email(invoice, template):
     """Collect the required data for sending emails."""
-    invoice = Invoice.objects.get(pk=invoice_pk)
-    recipient_email = invoice.order.user.email
+    recipient_email = invoice.order.user_email or invoice.order.user.email
```

Comment:
If I create a draft order with billing address and without email and customer? 
I'm not sure what should we do.

Reply:
Egzample:
![image](https://user-images.githubusercontent.com/43955230/83757267-83712600-a670-11ea-8551-42964cf8553b.png)

Reply:
For now we only allow even requesting an Invoice for an Order that is not draft but indeed if that transforms to order without `order.user_email` or `order.user.email` then we effectively have no way to know the email to send it to?
So I would add that check to SendInvoice mutation unless there is some other way we can get the email address?
Please advise, @maarcingebala maybe?

Reply:
@fowczarek discussed with Marcin, there is a requirement of adding the email in checkout so that's saved on `order.user_email`, otherwise we'll have the user so we can still get the email, so we don't have a case where both can be empty - current implementation should be okay.